x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticator.java

```diff
@@ -395,4 +405,18 @@ private void checkLifetimeRestrictions(Conditions conditions) {
     private void checkLifetimeRestrictions(SubjectConfirmationData subjectConfirmationData) {
         validateNotOnOrAfter(subjectConfirmationData.getNotOnOrAfter());
     }
+
+    private void warnOnReservedTermAttributeNames(Assertion assertion, List<Attribute> attributes) {
+        attributes.stream().map(Attribute::getName).forEach(attributeName -> {
+            if (RESERVED_TERM_ATTRIBUTE_NAMES.contains(attributeName)) {
+                logger.warn(
+                    "SAML assertion [{}] has attribute with name [{}]. This clashes with a reserved term. "
+                        + "Rename attribute to not clash with any of [{}]",
```

Comment:
The message says there is a clash, but didn't say what it entails by having a clash. I think we should say something along the line of "Attributes cannot be mapped if their names clash with special (or synthetic) attribute names". Please feel free to paraphrase.